types/twitter-autohook/index.d.ts

```diff
@@ -0,0 +1,43 @@
+// Type definitions for twitter-autohook 1.7
+// Project: https://github.com/twitterdev/autohook#readme
+// Definitions by: David Jiménez <https://github.com/dubisdev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import EventEmitter = require('events');
+
+export class Autohook extends EventEmitter {
+    constructor(configuration: {
```

Comment:
please review the configuration, it looks like being completely optional with all properties optional as well (with defualts):
https://github.com/twitterdev/autohook/blob/master/index.js#L118-L126

Reply:
Hi, I think they left defaults to ensure the `trim` method, but then the constructor throws error if they are empty, so I think they shouldn't be optional.

https://github.com/twitterdev/autohook/blob/master/index.js#L128-L133

Reply:
Yes, but I think that means all setup is required to be from environment, say on AZ devops setup, note that they are checking if only ENV vars are set, e.g. 'headers' is not validated at runtime, can be defaulted

Reply:
It's true, you're right. It's already solved, thanks for your review and for the work all your team does 🙌